In [1]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [2]:
import os
os.chdir('/content/drive/My Drive/ReInceptionE/')

In [3]:
print(os.getcwd())

/content/drive/My Drive/ReInceptionE


In [ ]:
%run InceptionE/main_train_InceptionE_WN18RR.py

datasets/FB15k-237/entity2vec.txt vector size:(14541, 100)
datasets/FB15k-237/relation2vec.txt vector size:(237, 100)
data num: 35070
data num: 40932
data num: 544230
init embedding from pretrained...
epoch: 0, lr: 1.8813804334965088e-08 ,total loss :20450.691319465637, time: 56.841806411743164
epoch: 1, lr: 4.0016961820470746e-05 ,total loss :13446.464308738708, time: 55.43017339706421
epoch: 2, lr: 8.001510983660652e-05 ,total loss :11029.194402694702, time: 55.36864876747131
epoch: 3, lr: 0.00012001325785274231 ,total loss :10212.793320178986, time: 55.35924768447876
number of test data: 35070
MR   MRR hit@1    hit@3   hit@10
239.9706871970345	0.3102490742740391	0.2262902765896778	0.34046193327630453	0.47744510978043914
number of test data: 40932
MR   MRR hit@1    hit@3   hit@10
256.36807387862797	0.3075187274374244	0.22261311443369491	0.3375354246066647	0.47561809830939117
epoch: 4, lr: 0.00016001140586887808 ,total loss :9723.849927902222, time: 159.39800763130188
epoch: 5, lr: 0.

KeyboardInterrupt: ignored

 datasets/WN18RR/entity2vec.txt vector size:(40943, 100)
datasets/WN18RR/relation2vec.txt vector size:(11, 100)
data num: 6068
data num: 6268
data num: 173670
init embedding from pretrained...
epoch: 0, lr: 5.894502225404845e-08 ,total loss :7421.678304672241, time: 35.10261416435242
epoch: 1, lr: 4.008261513275294e-05 ,total loss :5402.077360630035, time: 25.862781524658203
epoch: 2, lr: 8.010628524325184e-05 ,total loss :3937.6654076576233, time: 25.426547288894653
epoch: 3, lr: 0.00012012995535375073 ,total loss :3636.3236227035522, time: 25.62090754508972
number of test data: 6068
MR   MRR hit@1    hit@3   hit@10
1371.855141727093	0.314263904340971	0.21440342781806196	0.36980883322346736	0.4996704021094265
number of test data: 6268
MR   MRR hit@1    hit@3   hit@10
1251.1129546904913	0.3114623929279451	0.20979578813018507	0.3715698787492023	0.49617102744097
epoch: 4, lr: 0.0001601536254642496 ,total loss :3486.7668781280518, time: 77.2019031047821
epoch: 5, lr: 0.00015475618749999996 ,total loss :3332.462778568268, time: 25.690998554229736
epoch: 6, lr: 0.00014701837812499996 ,total loss :3206.9374108314514, time: 25.543790578842163
epoch: 7, lr: 0.00013966745921874994 ,total loss :3093.5551166534424, time: 25.55513310432434
epoch: 8, lr: 0.00013268408625781246 ,total loss :2999.0740995407104, time: 25.534153699874878
number of test data: 6068
MR   MRR hit@1    hit@3   hit@10
1505.6652933421226	0.345628975781557	0.244726433750824	0.40738299274884643	0.527027027027027
number of test data: 6268
MR   MRR hit@1    hit@3   hit@10
1375.360561582642	0.3471311858329029	0.248085513720485	0.40619017230376514	0.5242501595405233
epoch: 9, lr: 0.0001260498819449218 ,total loss :2916.3908693790436, time: 79.48712992668152
epoch: 10, lr: 0.00011974738784767575 ,total loss :2841.30207157135, time: 26.417852878570557
epoch: 11, lr: 0.00011376001845529194 ,total loss :2775.6699674129486, time: 25.705249547958374
epoch: 12, lr: 0.00010807201753252734 ,total loss :2716.5547647476196, time: 25.801384210586548
epoch: 13, lr: 0.00010266841665590097 ,total loss :2664.3899235725403, time: 26.051278829574585
number of test data: 6068
MR   MRR hit@1    hit@3   hit@10
1661.7560975609756	0.365039090768319	0.2668094924192485	0.4273236651285432	0.5420237310481213
number of test data: 6268
MR   MRR hit@1    hit@3   hit@10
1541.5641352903638	0.36532456770051264	0.2662731333758775	0.42661135928525845	0.5413209955328654
epoch: 14, lr: 9.75349958231059e-05 ,total loss :2612.114413499832, time: 75.03777718544006
epoch: 15, lr: 9.265824603195061e-05 ,total loss :2571.8900735378265, time: 25.930973052978516
epoch: 16, lr: 8.802533373035307e-05 ,total loss :2528.9184901714325, time: 25.845070838928223
epoch: 17, lr: 8.362406704383542e-05 ,total loss :2495.4705686569214, time: 25.69452214241028
epoch: 18, lr: 7.944286369164364e-05 ,total loss :2459.125930070877, time: 25.61576747894287
number of test data: 6068
MR   MRR hit@1    hit@3   hit@10
1871.5880026367831	0.37605871552276193	0.2794990112063283	0.4377059986816084	0.5468029004614371
number of test data: 6268
MR   MRR hit@1    hit@3   hit@10
1727.672144224633	0.3783862922218317	0.28095086151882576	0.43921506062539883	0.5483407785577536
epoch: 19, lr: 7.547072050706145e-05 ,total loss :2428.2459061145782, time: 74.44647359848022
epoch: 20, lr: 7.169718448170838e-05 ,total loss :2401.0414197444916, time: 26.043344259262085
epoch: 21, lr: 6.811232525762296e-05 ,total loss :2373.4717309474945, time: 26.161747932434082
epoch: 22, lr: 6.470670899474181e-05 ,total loss :2349.8283607959747, time: 26.334961652755737
epoch: 23, lr: 6.147137354500472e-05 ,total loss :2327.937908887863, time: 25.919071912765503
number of test data: 6068
MR   MRR hit@1    hit@3   hit@10
2014.4110085695452	0.38418256071812856	0.2897165458141068	0.4436387607119314	0.5484508899143046
number of test data: 6268
MR   MRR hit@1    hit@3   hit@10
1867.92102744097	0.3864262458241825	0.29132099553286533	0.4454371410338226	0.5512125079770261
epoch: 24, lr: 5.839780486775448e-05 ,total loss :2308.018297433853, time: 73.85907673835754
epoch: 25, lr: 5.5477914624366753e-05 ,total loss :2283.822327375412, time: 26.52553415298462
epoch: 26, lr: 5.270401889314841e-05 ,total loss :2267.192753791809, time: 25.911170959472656
epoch: 27, lr: 5.0068817948490985e-05 ,total loss :2253.5680623054504, time: 25.975656747817993
epoch: 28, lr: 6.634204312890623e-05 ,total loss :2242.3740718364716, time: 26.183736324310303
number of test data: 6068
MR   MRR hit@1    hit@3   hit@10
2096.1254119973632	0.39113404155712334	0.2992748846407383	0.4475939353988134	0.5515820698747528
number of test data: 6268
MR   MRR hit@1    hit@3   hit@10
1950.1967134652202	0.39272186559713884	0.2981812380344607	0.4502233567326101	0.5547223994894703
epoch: 29, lr: 6.30249409724609e-05 ,total loss :2221.6303606033325, time: 73.8801703453064
epoch: 30, lr: 0.0001 ,total loss :2217.2124214172363, time: 26.652384281158447
epoch: 31, lr: 9.5e-05 ,total loss :2194.3582689762115, time: 25.90099835395813
epoch: 32, lr: 9.025e-05 ,total loss :2174.7097136974335, time: 25.89640212059021
epoch: 33, lr: 8.573749999999999e-05 ,total loss :2153.100739955902, time: 25.966958045959473
number of test data: 6068
MR   MRR hit@1    hit@3   hit@10
2152.8744232036915	0.39855576431897904	0.3091628213579433	0.45319709953856296	0.5560316413974951
number of test data: 6268
MR   MRR hit@1    hit@3   hit@10
2008.57897255903	0.4022376955845714	0.31285896617740905	0.4569240587109126	0.5577536694320357
epoch: 34, lr: 8.1450625e-05 ,total loss :2128.800623893738, time: 73.93841314315796
epoch: 35, lr: 7.737809374999998e-05 ,total loss :2112.0889225006104, time: 26.06969428062439
epoch: 36, lr: 7.350918906249998e-05 ,total loss :2089.4908542633057, time: 25.84204077720642
epoch: 37, lr: 6.983372960937497e-05 ,total loss :2078.4624996185303, time: 25.953216314315796
epoch: 38, lr: 6.634204312890623e-05 ,total loss :2060.277174949646, time: 25.74696135520935
number of test data: 6068
MR   MRR hit@1    hit@3   hit@10
2168.859920896506	0.4064616551018052	0.32152274225444955	0.4574818721160185	0.5555372445616348
number of test data: 6268
MR   MRR hit@1    hit@3   hit@10
2025.3820995532865	0.4091438696555795	0.32227185705169115	0.461869814932993	0.5580727504786216
epoch: 39, lr: 6.30249409724609e-05 ,total loss :2043.0878405570984, time: 74.57220458984375
epoch: 40, lr: 0.0001 ,total loss :2044.7191362380981, time: 26.17118549346924
epoch: 41, lr: 9.5e-05 ,total loss :2024.5291240215302, time: 25.87341618537903
epoch: 42, lr: 9.025e-05 ,total loss :2006.2572932243347, time: 26.19960594177246
epoch: 43, lr: 8.573749999999999e-05 ,total loss :1994.6835701465607, time: 25.820393323898315
number of test data: 6068
MR   MRR hit@1    hit@3   hit@10
2181.473137771918	0.4131771523006849	0.33075148319050757	0.46110744891232697	0.5575148319050758
number of test data: 6268
MR   MRR hit@1    hit@3   hit@10
2039.8525845564773	0.41570906861387164	0.3329610721123165	0.46314613911933633	0.5587109125717933
epoch: 44, lr: 8.1450625e-05 ,total loss :1978.5190925598145, time: 74.27301669120789
epoch: 45, lr: 7.737809374999998e-05 ,total loss :1958.6416964530945, time: 26.137293100357056
epoch: 46, lr: 7.350918906249998e-05 ,total loss :1945.965470790863, time: 25.871662378311157
epoch: 47, lr: 6.983372960937497e-05 ,total loss :1931.213101387024, time: 25.92064929008484
epoch: 48, lr: 6.634204312890623e-05 ,total loss :1919.382265329361, time: 25.844666481018066
number of test data: 6068
MR   MRR hit@1    hit@3   hit@10
2172.545978905735	0.41948833245298667	0.3409690177982861	0.4648978246539222	0.5578444297956493
number of test data: 6268
MR   MRR hit@1    hit@3   hit@10
2037.3353541799618	0.42252780456831096	0.34317166560306317	0.467613273771538	0.5591895341416719
epoch: 49, lr: 6.30249409724609e-05 ,total loss :1909.4942979812622, time: 74.06010699272156
epoch: 50, lr: 0.0001 ,total loss :1906.95272231102, time: 26.198219299316406
epoch: 51, lr: 9.5e-05 ,total loss :1899.6097958087921, time: 26.138705492019653
epoch: 52, lr: 9.025e-05 ,total loss :1886.3021609783173, time: 26.085649490356445
epoch: 53, lr: 8.573749999999999e-05 ,total loss :1870.5054380893707, time: 26.05938482284546
number of test data: 6068
MR   MRR hit@1    hit@3   hit@10
2178.2389584706657	0.4244761843035167	0.34871456822676333	0.46704021094264997	0.5599868160843771
number of test data: 6268
MR   MRR hit@1    hit@3   hit@10
2034.7616464582004	0.42878576313507893	0.35354179961710275	0.47128270580727505	0.5598276962348436
epoch: 54, lr: 8.1450625e-05 ,total loss :1855.593428850174, time: 73.7865526676178
epoch: 55, lr: 7.737809374999998e-05 ,total loss :1840.8096404075623, time: 26.13109254837036
epoch: 56, lr: 7.350918906249998e-05 ,total loss :1830.1395218372345, time: 26.156121253967285
epoch: 57, lr: 6.983372960937497e-05 ,total loss :1822.1145870685577, time: 25.953484296798706
epoch: 58, lr: 6.634204312890623e-05 ,total loss :1807.7582650184631, time: 25.82148265838623
number of test data: 6068
MR   MRR hit@1    hit@3   hit@10
2168.6348055372446	0.4290083766782108	0.3561305207646671	0.46802900461437047	0.5580092287409361
number of test data: 6268
MR   MRR hit@1    hit@3   hit@10
2023.4191129546905	0.43416043193942777	0.36151882578174854	0.4738353541799617	0.5601467772814295
epoch: 59, lr: 6.30249409724609e-05 ,total loss :1801.8048174381256, time: 73.82783937454224
epoch: 60, lr: 0.0001 ,total loss :1810.2455122470856, time: 26.090762853622437
epoch: 61, lr: 9.5e-05 ,total loss :1798.8087413311005, time: 26.178336143493652
epoch: 62, lr: 9.025e-05 ,total loss :1784.2519979476929, time: 25.920592784881592
epoch: 63, lr: 8.573749999999999e-05 ,total loss :1774.3519637584686, time: 25.738269329071045
number of test data: 6068
MR   MRR hit@1    hit@3   hit@10
2159.6745220830585	0.4349862683244583	0.3661832564271589	0.4705009887936717	0.5576796308503625
number of test data: 6268
MR   MRR hit@1    hit@3   hit@10
2011.9513401403956	0.43831358162787554	0.368379068283344	0.4773452456924059	0.5631780472239949
epoch: 64, lr: 8.1450625e-05 ,total loss :1764.495932817459, time: 74.05199122428894
epoch: 65, lr: 7.737809374999998e-05 ,total loss :1752.5207755565643, time: 26.18273687362671
epoch: 66, lr: 7.350918906249998e-05 ,total loss :1744.5657737255096, time: 25.78513813018799
epoch: 67, lr: 6.983372960937497e-05 ,total loss :1731.9945001602173, time: 25.925785064697266
epoch: 68, lr: 6.634204312890623e-05 ,total loss :1719.9242177009583, time: 25.7693088054657
number of test data: 6068
MR   MRR hit@1    hit@3   hit@10
2149.892880685564	0.4368876779722801	0.3684904416611734	0.47214897824653923	0.557350032959789
number of test data: 6268
MR   MRR hit@1    hit@3   hit@10
2009.2204850031908	0.4423014076041984	0.3752393107849394	0.4779834077855775	0.5601467772814295
epoch: 69, lr: 6.30249409724609e-05 ,total loss :1724.015496969223, time: 74.04725313186646
epoch: 70, lr: 0.0001 ,total loss :1727.7482476234436, time: 26.044015884399414
epoch: 71, lr: 9.5e-05 ,total loss :1716.5597033500671, time: 25.8694429397583
epoch: 72, lr: 9.025e-05 ,total loss :1708.357054233551, time: 26.241410493850708
epoch: 73, lr: 8.573749999999999e-05 ,total loss :1696.8672518730164, time: 25.80660581588745
number of test data: 6068
MR   MRR hit@1    hit@3   hit@10
2135.0037903757416	0.4399600130625531	0.37376400791034936	0.47396176664469347	0.5601516150296638
number of test data: 6268
MR   MRR hit@1    hit@3   hit@10
1993.777281429483	0.445916762340667	0.38066368857689853	0.4795788130185067	0.5619017230376515
epoch: 74, lr: 8.1450625e-05 ,total loss :1686.8350360393524, time: 74.29079937934875
epoch: 75, lr: 7.737809374999998e-05 ,total loss :1677.6540355682373, time: 26.45509386062622
epoch: 76, lr: 7.350918906249998e-05 ,total loss :1667.5154247283936, time: 25.821426153182983
epoch: 77, lr: 6.983372960937497e-05 ,total loss :1663.9631118774414, time: 26.27608060836792
epoch: 78, lr: 6.634204312890623e-05 ,total loss :1655.4778780937195, time: 25.854166984558105
number of test data: 6068
MR   MRR hit@1    hit@3   hit@10
2124.7165458141067	0.44648739948657234	0.3848055372445616	0.4752801582069875	0.557350032959789
number of test data: 6268
MR   MRR hit@1    hit@3   hit@10
1980.8888002552649	0.44916006833071276	0.38656668793873644	0.48117421825143586	0.5625398851308232
epoch: 79, lr: 6.30249409724609e-05 ,total loss :1649.8759951591492, time: 74.9181752204895
epoch: 80, lr: 0.0001 ,total loss :1657.019244670868, time: 26.32801342010498
epoch: 81, lr: 9.5e-05 ,total loss :1647.8943219184875, time: 25.97248888015747
epoch: 82, lr: 9.025e-05 ,total loss :1645.1603999137878, time: 26.57644748687744
epoch: 83, lr: 8.573749999999999e-05 ,total loss :1635.619381904602, time: 25.962961673736572
number of test data: 6068
MR   MRR hit@1    hit@3   hit@10
2108.2162162162163	0.44940098882933116	0.39007910349373764	0.4767633487145682	0.5588332234673699
number of test data: 6268
MR   MRR hit@1    hit@3   hit@10
1964.5031908104659	0.4525769011633724	0.39183152520740266	0.4835673261008296	0.5619017230376515
epoch: 84, lr: 8.1450625e-05 ,total loss :1627.719334602356, time: 74.07636070251465
epoch: 85, lr: 7.737809374999998e-05 ,total loss :1620.1324949264526, time: 26.49726891517639
epoch: 86, lr: 7.350918906249998e-05 ,total loss :1611.0600082874298, time: 25.862055778503418
epoch: 87, lr: 6.983372960937497e-05 ,total loss :1606.8601651191711, time: 25.795505046844482
epoch: 88, lr: 6.634204312890623e-05 ,total loss :1598.0733160972595, time: 25.77459192276001
number of test data: 6068
MR   MRR hit@1    hit@3   hit@10
2097.538892551088	0.4511624476035647	0.39255108767303887	0.476928147659855	0.5617996044825313
number of test data: 6268
MR   MRR hit@1    hit@3   hit@10
1946.7996171027442	0.45431850411743957	0.3955009572431398	0.48181238034460755	0.5615826419910658
epoch: 89, lr: 6.30249409724609e-05 ,total loss :1590.9184824228287, time: 74.19624519348145
epoch: 90, lr: 0.0001 ,total loss :1601.3111919164658, time: 25.938345670700073
epoch: 91, lr: 9.5e-05 ,total loss :1594.2260053157806, time: 25.699875831604004
epoch: 92, lr: 9.025e-05 ,total loss :1583.6803455352783, time: 25.78909993171692
epoch: 93, lr: 8.573749999999999e-05 ,total loss :1582.9123150110245, time: 25.78670334815979
number of test data: 6068
MR   MRR hit@1    hit@3   hit@10
2077.7369808833223	0.4543403621879873	0.3974950560316414	0.47857613711272246	0.5599868160843771
number of test data: 6268
MR   MRR hit@1    hit@3   hit@10
1939.4022016592214	0.4571896522766697	0.40044671346522015	0.4843650287172942	0.5622208040842374
epoch: 94, lr: 8.1450625e-05 ,total loss :1572.4986106157303, time: 76.82679677009583
epoch: 95, lr: 7.737809374999998e-05 ,total loss :1566.4271693229675, time: 26.038867235183716
epoch: 96, lr: 7.350918906249998e-05 ,total loss :1559.2546560764313, time: 26.10054588317871
epoch: 97, lr: 6.983372960937497e-05 ,total loss :1554.6468218564987, time: 25.99000310897827
epoch: 98, lr: 6.634204312890623e-05 ,total loss :1548.1755838394165, time: 26.35217833518982
number of test data: 6068
MR   MRR hit@1    hit@3   hit@10
2063.282300593276	0.4567131347903813	0.4021094264996704	0.478905735003296	0.5604812129202373
number of test data: 6268
MR   MRR hit@1    hit@3   hit@10
1921.8895979578813	0.4599203377424742	0.40507338864071474	0.48404594767070835	0.561104020421187
epoch: 99, lr: 6.30249409724609e-05 ,total loss :1541.7435653209686, time: 76.7378032207489
head ranking ....
MR   MRR hit@1    hit@3   hit@10
3040.61774090619	0.4146934662616443	0.36821952776005107	0.43426930440331846	0.49393746011486916
tail ranking ....
MR   MRR hit@1    hit@3   hit@10
803.1611359285258	0.5051472092277901	0.4419272495213784	0.5338225909380983	0.6282705807275047
Total ranking ...
MR   MRR hit@1    hit@3   hit@10
1921.889438417358	0.4599203377447173	0.40507338864071474	0.48404594767070835	0.561104020421187

datasets/WN18RR/entity2vec.txt vector size:(40943, 100)
datasets/WN18RR/relation2vec.txt vector size:(11, 100)
data num: 6068
data num: 6268
data num: 173670
init embedding from pretrained...
epoch: 0, lr: 5.894502225404845e-08 ,total loss :7372.539793014526, time: 26.775640964508057
epoch: 1, lr: 4.008261513275294e-05 ,total loss :5266.292137145996, time: 12.35303783416748
epoch: 2, lr: 8.010628524325184e-05 ,total loss :3881.138202190399, time: 12.052516460418701
epoch: 3, lr: 0.00012012995535375073 ,total loss :3610.813416004181, time: 12.013288497924805
number of test data: 6068
MR   MRR hit@1    hit@3   hit@10
1344.5922874093606	0.3156111668497273	0.21704021094264997	0.3666776532630191	0.5016479894528675
number of test data: 6268
MR   MRR hit@1    hit@3   hit@10
1219.8616783663051	0.31201602064533096	0.209955328653478	0.3696553924696873	0.4990427568602425
epoch: 4, lr: 0.0001601536254642496 ,total loss :3449.0221643447876, time: 61.090535163879395
epoch: 5, lr: 0.00015475618749999996 ,total loss :3302.614380836487, time: 12.020620822906494
epoch: 6, lr: 0.00014701837812499996 ,total loss :3173.6337842941284, time: 12.083122968673706
epoch: 7, lr: 0.00013966745921874994 ,total loss :3071.614661693573, time: 12.77562689781189
epoch: 8, lr: 0.00013268408625781246 ,total loss :2982.2432918548584, time: 12.223746061325073
number of test data: 6068
MR   MRR hit@1    hit@3   hit@10
1494.926994067238	0.347339719384675	0.2470336189848385	0.4067237969676994	0.5299934080421885
number of test data: 6268
MR   MRR hit@1    hit@3   hit@10
1369.1025845564773	0.34553628226777505	0.244256541161455	0.4055520102105935	0.5288768347160179
epoch: 9, lr: 0.0001260498819449218 ,total loss :2897.2186465263367, time: 60.70290565490723
epoch: 10, lr: 0.00011974738784767575 ,total loss :2821.94899559021, time: 12.69461441040039
epoch: 11, lr: 0.00011376001845529194 ,total loss :2763.182557106018, time: 12.171558618545532
epoch: 12, lr: 0.00010807201753252734 ,total loss :2703.0625970363617, time: 12.198413848876953
epoch: 13, lr: 0.00010266841665590097 ,total loss :2649.185869216919, time: 12.755170106887817
number of test data: 6068
MR   MRR hit@1    hit@3   hit@10
1686.6481542518127	0.36342766267796467	0.26367831245880025	0.42320369149637443	0.5388925510876731
number of test data: 6268
MR   MRR hit@1    hit@3   hit@10
1525.719527760051	0.3650373406351425	0.2645181876196554	0.42549457562220805	0.5430759412890874
epoch: 14, lr: 9.75349958231059e-05 ,total loss :2601.8193850517273, time: 60.36698603630066
epoch: 15, lr: 9.265824603195061e-05 ,total loss :2559.0747497081757, time: 12.196939468383789
epoch: 16, lr: 8.802533373035307e-05 ,total loss :2524.5756266117096, time: 12.26591420173645
epoch: 17, lr: 8.362406704383542e-05 ,total loss :2486.9175963401794, time: 12.18994688987732
epoch: 18, lr: 7.944286369164364e-05 ,total loss :2448.7997307777405, time: 12.20598292350769
number of test data: 6068
MR   MRR hit@1    hit@3   hit@10
1883.1320039551747	0.37532912609092073	0.27702702702702703	0.4372116018457482	0.54614370468029
number of test data: 6268
MR   MRR hit@1    hit@3   hit@10
1715.1975111678366	0.37645833359505904	0.2774409700063816	0.4395341416719847	0.5504148053605615
epoch: 19, lr: 7.547072050706145e-05 ,total loss :2421.389966249466, time: 60.835087060928345
epoch: 20, lr: 7.169718448170838e-05 ,total loss :2392.34113574028, time: 12.249450922012329
epoch: 21, lr: 6.811232525762296e-05 ,total loss :2366.9059607982635, time: 12.237390518188477
epoch: 22, lr: 6.470670899474181e-05 ,total loss :2345.80335521698, time: 12.735383749008179
epoch: 23, lr: 6.147137354500472e-05 ,total loss :2327.3178548812866, time: 12.240758180618286
number of test data: 6068
MR   MRR hit@1    hit@3   hit@10
2024.0847066578774	0.38345914483879767	0.28790375741595253	0.44330916282135796	0.548121292023731
number of test data: 6268
MR   MRR hit@1    hit@3   hit@10
1856.411774090619	0.3860351079675599	0.29020421186981493	0.44751116783663053	0.5536056158264199
epoch: 24, lr: 5.839780486775448e-05 ,total loss :2300.981899499893, time: 60.73941731452942
epoch: 25, lr: 5.5477914624366753e-05 ,total loss :2282.587423801422, time: 12.22985577583313
epoch: 26, lr: 5.270401889314841e-05 ,total loss :2259.5069711208344, time: 12.196321249008179
epoch: 27, lr: 5.0068817948490985e-05 ,total loss :2245.0161170959473, time: 12.298068523406982
epoch: 28, lr: 6.634204312890623e-05 ,total loss :2243.481612920761, time: 12.720389127731323
number of test data: 6068
MR   MRR hit@1    hit@3   hit@10
2111.7935069215555	0.3908950982084341	0.29812129202373105	0.4507251153592617	0.5507580751483191
number of test data: 6268
MR   MRR hit@1    hit@3   hit@10
1930.827377153797	0.39263937146868455	0.2992980216975112	0.45165922144224635	0.5555201021059349
epoch: 29, lr: 6.30249409724609e-05 ,total loss :2223.4595119953156, time: 60.15738606452942
epoch: 30, lr: 0.0001 ,total loss :2226.4048266410828, time: 12.155131101608276
epoch: 31, lr: 9.5e-05 ,total loss :2196.0552620887756, time: 12.650934934616089
epoch: 32, lr: 9.025e-05 ,total loss :2168.4856021404266, time: 12.204416751861572
epoch: 33, lr: 8.573749999999999e-05 ,total loss :2148.4798305034637, time: 12.256922483444214
number of test data: 6068
MR   MRR hit@1    hit@3   hit@10
2177.547791694133	0.3985009956808561	0.3078444297956493	0.45484508899143045	0.5538892551087673
number of test data: 6268
MR   MRR hit@1    hit@3   hit@10
1996.615985960434	0.4010700569965964	0.31078493937460117	0.4562858966177409	0.5555201021059349
epoch: 34, lr: 8.1450625e-05 ,total loss :2129.392354488373, time: 60.47421479225159
epoch: 35, lr: 7.737809374999998e-05 ,total loss :2104.450207233429, time: 12.198801517486572
epoch: 36, lr: 7.350918906249998e-05 ,total loss :2087.8804490566254, time: 12.300015211105347
epoch: 37, lr: 6.983372960937497e-05 ,total loss :2072.157613992691, time: 12.526342868804932
epoch: 38, lr: 6.634204312890623e-05 ,total loss :2058.492812871933, time: 12.273107528686523
number of test data: 6068
MR   MRR hit@1    hit@3   hit@10
2199.7420896506264	0.4068379795516226	0.3206987475280158	0.46110744891232697	0.5537244561634805
number of test data: 6268
MR   MRR hit@1    hit@3   hit@10
2021.412252712189	0.4077336453829376	0.3211550733886407	0.4623484365028717	0.5559987236758137
epoch: 39, lr: 6.30249409724609e-05 ,total loss :2040.5017058849335, time: 60.29976487159729
epoch: 40, lr: 0.0001 ,total loss :2045.7853729724884, time: 12.609337091445923
epoch: 41, lr: 9.5e-05 ,total loss :2023.8093793392181, time: 12.182549476623535
epoch: 42, lr: 9.025e-05 ,total loss :2008.6013023853302, time: 12.2633957862854
epoch: 43, lr: 8.573749999999999e-05 ,total loss :1992.1982340812683, time: 12.59683609008789
number of test data: 6068
MR   MRR hit@1    hit@3   hit@10
2217.4724785761373	0.4133728466329611	0.3309162821357943	0.4634146341463415	0.5540540540540541
number of test data: 6268
MR   MRR hit@1    hit@3   hit@10
2033.7732929164008	0.4161139542603086	0.3329610721123165	0.46490108487555837	0.5598276962348436
epoch: 44, lr: 8.1450625e-05 ,total loss :1978.2882463932037, time: 60.21758580207825
epoch: 45, lr: 7.737809374999998e-05 ,total loss :1957.400588274002, time: 12.20456337928772
epoch: 46, lr: 7.350918906249998e-05 ,total loss :1947.2871701717377, time: 12.256118535995483
epoch: 47, lr: 6.983372960937497e-05 ,total loss :1929.5885605812073, time: 12.198925971984863
epoch: 48, lr: 6.634204312890623e-05 ,total loss :1921.5207810401917, time: 12.27785849571228
number of test data: 6068
MR   MRR hit@1    hit@3   hit@10
2220.5009887936717	0.4197875350683255	0.34080421885299933	0.4673698088332235	0.5537244561634805
number of test data: 6268
MR   MRR hit@1    hit@3   hit@10
2038.6257179323547	0.42026578070217196	0.3398213146139119	0.4685705169112955	0.5595086151882578
epoch: 49, lr: 6.30249409724609e-05 ,total loss :1910.5373299121857, time: 60.47156381607056
epoch: 50, lr: 0.0001 ,total loss :1914.3584866523743, time: 12.255322217941284
epoch: 51, lr: 9.5e-05 ,total loss :1898.2898614406586, time: 12.209957599639893
epoch: 52, lr: 9.025e-05 ,total loss :1881.4039895534515, time: 12.306134223937988
epoch: 53, lr: 8.573749999999999e-05 ,total loss :1876.5487353801727, time: 12.342048645019531
number of test data: 6068
MR   MRR hit@1    hit@3   hit@10
2214.7564271588662	0.42459200654913476	0.34838497033618987	0.46901779828609097	0.5548780487804879
number of test data: 6268
MR   MRR hit@1    hit@3   hit@10
2032.9653797064454	0.42621259753722585	0.3485960433950223	0.47064454371410336	0.5607849393746012
epoch: 54, lr: 8.1450625e-05 ,total loss :1857.6717386245728, time: 61.1210355758667
epoch: 55, lr: 7.737809374999998e-05 ,total loss :1845.7149770259857, time: 12.319844007492065
epoch: 56, lr: 7.350918906249998e-05 ,total loss :1829.8935706615448, time: 12.299944400787354
epoch: 57, lr: 6.983372960937497e-05 ,total loss :1817.8949418067932, time: 12.838603019714355
epoch: 58, lr: 6.634204312890623e-05 ,total loss :1809.4425749778748, time: 12.222149133682251
number of test data: 6068
MR   MRR hit@1    hit@3   hit@10
2212.0481212920236	0.430010805812113	0.3571193144363876	0.47099538562953197	0.555372445616348
number of test data: 6268
MR   MRR hit@1    hit@3   hit@10
2026.5861518825782	0.43236156940170895	0.3591257179323548	0.47415443522654754	0.5598276962348436
epoch: 59, lr: 6.30249409724609e-05 ,total loss :1799.6139175891876, time: 60.481799840927124
epoch: 60, lr: 0.0001 ,total loss :1807.3869078159332, time: 12.588285207748413
epoch: 61, lr: 9.5e-05 ,total loss :1797.9500224590302, time: 12.242641687393188
epoch: 62, lr: 9.025e-05 ,total loss :1785.4732275009155, time: 12.273864507675171
epoch: 63, lr: 8.573749999999999e-05 ,total loss :1773.140771150589, time: 12.688093423843384
number of test data: 6068
MR   MRR hit@1    hit@3   hit@10
2195.4049110085693	0.43548858441638194	0.36552406064601184	0.47363216875411995	0.5527356624917601
number of test data: 6268
MR   MRR hit@1    hit@3   hit@10
2017.405711550734	0.43631976491561975	0.3659859604339502	0.4746330567964263	0.5607849393746012
epoch: 64, lr: 8.1450625e-05 ,total loss :1763.1619334220886, time: 60.734715938568115
epoch: 65, lr: 7.737809374999998e-05 ,total loss :1757.1227355003357, time: 12.237033605575562
epoch: 66, lr: 7.350918906249998e-05 ,total loss :1743.1225113868713, time: 12.54169750213623
epoch: 67, lr: 6.983372960937497e-05 ,total loss :1731.9375824928284, time: 12.360631227493286
epoch: 68, lr: 6.634204312890623e-05 ,total loss :1724.8252820968628, time: 12.78809666633606
number of test data: 6068
MR   MRR hit@1    hit@3   hit@10
2192.913315754779	0.43984747740763985	0.3734344100197759	0.47495056031641397	0.5555372445616348
number of test data: 6268
MR   MRR hit@1    hit@3   hit@10
2015.9267708998086	0.44068342355084483	0.3723675813656669	0.47606892150606256	0.5606253988513082
epoch: 69, lr: 6.30249409724609e-05 ,total loss :1715.4683220386505, time: 60.88982605934143
epoch: 70, lr: 0.0001 ,total loss :1725.439952135086, time: 12.278160572052002
epoch: 71, lr: 9.5e-05 ,total loss :1716.7034301757812, time: 12.577333450317383
epoch: 72, lr: 9.025e-05 ,total loss :1708.3782694339752, time: 12.275899887084961
epoch: 73, lr: 8.573749999999999e-05 ,total loss :1701.5110795497894, time: 12.280691385269165
number of test data: 6068
MR   MRR hit@1    hit@3   hit@10
2179.6407382992747	0.44320257158128806	0.37870797626895186	0.4752801582069875	0.5583388266315096
number of test data: 6268
MR   MRR hit@1    hit@3   hit@10
2002.525526483727	0.44414474343619825	0.3774728781110402	0.47925973197192084	0.5636566687938737
epoch: 74, lr: 8.1450625e-05 ,total loss :1687.1423766613007, time: 60.84562969207764
epoch: 75, lr: 7.737809374999998e-05 ,total loss :1679.6683979034424, time: 12.292809247970581
epoch: 76, lr: 7.350918906249998e-05 ,total loss :1672.6327857971191, time: 12.242287158966064
epoch: 77, lr: 6.983372960937497e-05 ,total loss :1663.3919013738632, time: 12.546772241592407
epoch: 78, lr: 6.634204312890623e-05 ,total loss :1655.3700723648071, time: 12.297794103622437
number of test data: 6068
MR   MRR hit@1    hit@3   hit@10
2165.3175675675675	0.4476413697191429	0.3859591298615689	0.4792353328938695	0.5583388266315096
number of test data: 6268
MR   MRR hit@1    hit@3   hit@10
1984.82897255903	0.44648316398360355	0.38273771537970647	0.4784620293554563	0.5622208040842374
epoch: 79, lr: 6.30249409724609e-05 ,total loss :1646.829153060913, time: 61.28724455833435
epoch: 80, lr: 0.0001 ,total loss :1657.1944785118103, time: 12.242725610733032
epoch: 81, lr: 9.5e-05 ,total loss :1650.310581445694, time: 12.217894077301025
epoch: 82, lr: 9.025e-05 ,total loss :1641.3368599414825, time: 12.655303955078125
epoch: 83, lr: 8.573749999999999e-05 ,total loss :1631.8962916135788, time: 12.311625242233276
number of test data: 6068
MR   MRR hit@1    hit@3   hit@10
2160.778345418589	0.44935374182183135	0.3887607119314436	0.47841133816743575	0.5583388266315096
number of test data: 6268
MR   MRR hit@1    hit@3   hit@10
1977.8604020421187	0.4512181195518252	0.3902361199744735	0.48085513720485	0.5617421825143587
epoch: 84, lr: 8.1450625e-05 ,total loss :1621.4697937965393, time: 60.973170042037964
epoch: 85, lr: 7.737809374999998e-05 ,total loss :1615.6844799518585, time: 12.581024169921875
epoch: 86, lr: 7.350918906249998e-05 ,total loss :1610.4185700416565, time: 12.262048244476318
epoch: 87, lr: 6.983372960937497e-05 ,total loss :1605.1165229082108, time: 12.344879388809204
epoch: 88, lr: 6.634204312890623e-05 ,total loss :1595.3393944501877, time: 12.784520626068115
number of test data: 6068
MR   MRR hit@1    hit@3   hit@10
2146.1946275543837	0.4533135124231201	0.39502307185234015	0.48137771918259725	0.5566908371786421
number of test data: 6268
MR   MRR hit@1    hit@3   hit@10
1960.1518825781748	0.4539521281927922	0.39470325462667516	0.48213146139119334	0.5639757498404595
epoch: 89, lr: 6.30249409724609e-05 ,total loss :1586.5025782585144, time: 60.933953523635864
epoch: 90, lr: 0.0001 ,total loss :1597.5986775159836, time: 12.3111891746521
epoch: 91, lr: 9.5e-05 ,total loss :1592.6707170009613, time: 12.265426874160767
epoch: 92, lr: 9.025e-05 ,total loss :1585.8182580471039, time: 12.207111597061157
epoch: 93, lr: 8.573749999999999e-05 ,total loss :1580.3559453487396, time: 12.60137414932251
number of test data: 6068
MR   MRR hit@1    hit@3   hit@10
2128.4423203691495	0.4534516773148066	0.3961766644693474	0.48038892551087675	0.5568556361239289
number of test data: 6268
MR   MRR hit@1    hit@3   hit@10
1943.2208040842374	0.455644437646035	0.3978940650925335	0.48085513720485	0.5615826419910658
epoch: 94, lr: 8.1450625e-05 ,total loss :1574.2890754938126, time: 60.59681487083435
epoch: 95, lr: 7.737809374999998e-05 ,total loss :1569.820004940033, time: 12.267749547958374
epoch: 96, lr: 7.350918906249998e-05 ,total loss :1556.7505170106888, time: 12.19630241394043
epoch: 97, lr: 6.983372960937497e-05 ,total loss :1552.4937781095505, time: 12.32668161392212
epoch: 98, lr: 6.634204312890623e-05 ,total loss :1546.8658484220505, time: 12.613103151321411
number of test data: 6068
MR   MRR hit@1    hit@3   hit@10
2117.533289386948	0.4557648058869429	0.40029663810151617	0.48203691496374423	0.5570204350692155
number of test data: 6268
MR   MRR hit@1    hit@3   hit@10
1930.8315252074026	0.45673262027047384	0.3998085513720485	0.4797383535417996	0.5625398851308232
epoch: 99, lr: 6.30249409724609e-05 ,total loss :1542.0793225765228, time: 60.6528046131134
head ranking ....
MR   MRR hit@1    hit@3   hit@10
3032.896617740906	0.4130325435117809	0.3653477983407786	0.4310784939374601	0.49680918953414166
tail ranking ....
MR   MRR hit@1    hit@3   hit@10
828.7654754307595	0.5004326970456504	0.43426930440331846	0.5283982131461391	0.6282705807275047
Total ranking ...
MR   MRR hit@1    hit@3   hit@10
1930.8310465858328	0.45673262027871536	0.3998085513720485	0.4797383535417996	0.5625398851308232

datasets/FB15k-237/entity2vec.txt vector size:(14541, 100)
datasets/FB15k-237/relation2vec.txt vector size:(237, 100)
data num: 35070
data num: 40932
data num: 544230
init embedding from pretrained...
epoch: 0, lr: 9.406902167482544e-09 ,total loss :20471.382643699646, time: 41.254337549209595
epoch: 1, lr: 2.0008480910235373e-05 ,total loss :14819.940429210663, time: 26.672318935394287
epoch: 2, lr: 4.000755491830326e-05 ,total loss :11813.773085594177, time: 26.80405592918396
epoch: 3, lr: 6.0006628926371155e-05 ,total loss :10773.712572097778, time: 26.95809006690979
number of test data: 35070
MR   MRR hit@1    hit@3   hit@10
261.719503849444	0.301422790784323	0.2197034502423724	0.3289136013686912	0.46441402908468776
number of test data: 40932
MR   MRR hit@1    hit@3   hit@10
275.05553112479237	0.2991946878635542	0.2166764389719535	0.3266393042118636	0.4635981628066061
epoch: 4, lr: 8.000570293443904e-05 ,total loss :10188.883901119232, time: 123.75938415527344
epoch: 5, lr: 7.737809374999998e-05 ,total loss :9768.080529689789, time: 26.844354391098022
epoch: 6, lr: 7.350918906249998e-05 ,total loss :9485.65289902687, time: 26.8228919506073
epoch: 7, lr: 6.983372960937497e-05 ,total loss :9274.714988708496, time: 26.905499935150146
epoch: 8, lr: 6.634204312890623e-05 ,total loss :9106.600113630295, time: 26.929121017456055
number of test data: 35070
MR   MRR hit@1    hit@3   hit@10
234.1772740233818	0.3231541261507365	0.23669803250641575	0.35548902195608784	0.49412603364699176
number of test data: 40932
MR   MRR hit@1    hit@3   hit@10
250.49377015537965	0.3230181264859441	0.2375891722857422	0.35353757451382783	0.4923776018762826
epoch: 9, lr: 6.30249409724609e-05 ,total loss :8982.535757541656, time: 122.77733969688416
epoch: 10, lr: 5.987369392383787e-05 ,total loss :8870.073270559311, time: 26.9075608253479
epoch: 11, lr: 5.688000922764597e-05 ,total loss :8771.551979064941, time: 26.834763288497925
epoch: 12, lr: 5.403600876626367e-05 ,total loss :8691.92762684822, time: 26.89533281326294
epoch: 13, lr: 5.1334208327950485e-05 ,total loss :8618.322667837143, time: 26.962395668029785
number of test data: 35070
MR   MRR hit@1    hit@3   hit@10
238.73262047333904	0.32593982292059437	0.23972055888223553	0.3564585115483319	0.49860279441117766
number of test data: 40932
MR   MRR hit@1    hit@3   hit@10
254.58299130264828	0.3237561628708596	0.23722271083748656	0.35588292778266395	0.49496726277728914
epoch: 14, lr: 8.1450625e-05 ,total loss :8619.309203386307, time: 123.44035196304321
epoch: 15, lr: 7.737809374999998e-05 ,total loss :8541.080408334732, time: 26.82077121734619
epoch: 16, lr: 7.350918906249998e-05 ,total loss :8456.243744373322, time: 26.814555883407593
epoch: 17, lr: 6.983372960937497e-05 ,total loss :8390.159287929535, time: 26.967247009277344
epoch: 18, lr: 6.634204312890623e-05 ,total loss :8333.5224609375, time: 27.02310013771057
number of test data: 35070
MR   MRR hit@1    hit@3   hit@10
239.12072996863415	0.3302012037697744	0.24208725406330198	0.3635585970915312	0.5052181351582549
number of test data: 40932
MR   MRR hit@1    hit@3   hit@10
256.8800449526043	0.32831571700355244	0.24005667937066355	0.3610866803478941	0.5023453532688361
epoch: 19, lr: 6.30249409724609e-05 ,total loss :8277.391028642654, time: 122.31201720237732
epoch: 20, lr: 0.0001 ,total loss :8312.622517347336, time: 26.86277151107788
epoch: 21, lr: 9.5e-05 ,total loss :8252.850625514984, time: 26.871142148971558
epoch: 22, lr: 9.025e-05 ,total loss :8193.174097299576, time: 26.939659595489502
epoch: 23, lr: 8.573749999999999e-05 ,total loss :8138.933686733246, time: 26.952975034713745
number of test data: 35070
MR   MRR hit@1    hit@3   hit@10
246.98018249215855	0.33119466126107794	0.24274308525805532	0.36427145708582837	0.507670373538637
number of test data: 40932
MR   MRR hit@1    hit@3   hit@10
265.3884979966774	0.32936734333836704	0.24115606371543047	0.36323658751099386	0.5038111990618587
epoch: 24, lr: 8.1450625e-05 ,total loss :8091.267551660538, time: 122.39583277702332
epoch: 25, lr: 7.737809374999998e-05 ,total loss :8048.207417011261, time: 26.861629009246826
epoch: 26, lr: 7.350918906249998e-05 ,total loss :8003.030150413513, time: 26.894537687301636
epoch: 27, lr: 6.983372960937497e-05 ,total loss :7966.655844926834, time: 26.870483875274658
epoch: 28, lr: 6.634204312890623e-05 ,total loss :7925.801511287689, time: 26.989197254180908
number of test data: 35070
MR   MRR hit@1    hit@3   hit@10
251.24066153407472	0.3321151979801235	0.24439692044482464	0.36407185628742517	0.5096948959224409
number of test data: 40932
MR   MRR hit@1    hit@3   hit@10
268.8778461839148	0.328558432833766	0.23988566402814424	0.3620883416397928	0.5056679370663539
epoch: 29, lr: 6.30249409724609e-05 ,total loss :7899.838988304138, time: 122.53370475769043
epoch: 30, lr: 0.0001 ,total loss :7952.05059838295, time: 26.897104740142822
epoch: 31, lr: 9.5e-05 ,total loss :7921.126888990402, time: 26.91365361213684
epoch: 32, lr: 9.025e-05 ,total loss :7882.226299524307, time: 26.93436312675476
epoch: 33, lr: 8.573749999999999e-05 ,total loss :7849.398884773254, time: 26.985498189926147
number of test data: 35070
MR   MRR hit@1    hit@3   hit@10
255.84590818363273	0.33029013938547425	0.24157399486740805	0.3634445394924437	0.507442258340462
number of test data: 40932
MR   MRR hit@1    hit@3   hit@10
273.40706537672236	0.32762600774948714	0.2389572950258966	0.36142871103293267	0.5044708296687188
epoch: 34, lr: 8.1450625e-05 ,total loss :7815.840815544128, time: 122.22309112548828
epoch: 35, lr: 7.737809374999998e-05 ,total loss :7786.416485786438, time: 27.10215425491333
epoch: 36, lr: 7.350918906249998e-05 ,total loss :7758.377496957779, time: 27.038123846054077
epoch: 37, lr: 6.983372960937497e-05 ,total loss :7726.7490339279175, time: 26.991801261901855
epoch: 38, lr: 6.634204312890623e-05 ,total loss :7706.393805265427, time: 27.066065311431885
number of test data: 35070
MR   MRR hit@1    hit@3   hit@10
261.4320501853436	0.3303642416710404	0.24268605645851155	0.3628172226974622	0.5079840319361277
number of test data: 40932
MR   MRR hit@1    hit@3   hit@10
279.62415713866903	0.32696589002904036	0.2382732336558194	0.3600850190559953	0.5047884295905404
epoch: 39, lr: 6.30249409724609e-05 ,total loss :7680.603162050247, time: 122.60364413261414
epoch: 40, lr: 0.0001 ,total loss :7744.476213932037, time: 26.85276484489441
epoch: 41, lr: 9.5e-05 ,total loss :7720.893988132477, time: 26.871820211410522
epoch: 42, lr: 9.025e-05 ,total loss :7692.71869969368, time: 26.975997924804688
epoch: 43, lr: 8.573749999999999e-05 ,total loss :7672.596019029617, time: 27.010956287384033
number of test data: 35070
MR   MRR hit@1    hit@3   hit@10
268.0751354433989	0.330077912232234	0.24208725406330198	0.36333048189335615	0.5067579127459367
number of test data: 40932
MR   MRR hit@1    hit@3   hit@10
285.8091957392749	0.3282812263545558	0.2408384637936089	0.3604270497410339	0.5032737222710838
epoch: 44, lr: 8.1450625e-05 ,total loss :7643.430644989014, time: 122.60345888137817
epoch: 45, lr: 7.737809374999998e-05 ,total loss :7625.213434934616, time: 26.825345277786255
epoch: 46, lr: 7.350918906249998e-05 ,total loss :7596.348607540131, time: 26.912899255752563
epoch: 47, lr: 6.983372960937497e-05 ,total loss :7575.30563211441, time: 26.980920791625977
epoch: 48, lr: 6.634204312890623e-05 ,total loss :7553.213632822037, time: 27.046489000320435
number of test data: 35070
MR   MRR hit@1    hit@3   hit@10
268.4571998859424	0.33067434484580555	0.24274308525805532	0.36313088109495295	0.5088394639292843
number of test data: 40932
MR   MRR hit@1    hit@3   hit@10
285.8275676732141	0.32777901342157506	0.2392993257109352	0.36186846477083945	0.5036401837193394
epoch: 49, lr: 6.30249409724609e-05 ,total loss :7535.9842739105225, time: 122.43098711967468
epoch: 50, lr: 0.0001 ,total loss :7598.9117612838745, time: 26.957274436950684
epoch: 51, lr: 9.5e-05 ,total loss :7587.000791549683, time: 26.923375368118286
epoch: 52, lr: 9.025e-05 ,total loss :7566.552240371704, time: 26.935760259628296
epoch: 53, lr: 8.573749999999999e-05 ,total loss :7545.359529733658, time: 27.00220012664795
number of test data: 35070
MR   MRR hit@1    hit@3   hit@10
275.7370972341032	0.32823608371930724	0.23989164528086684	0.36102081551183346	0.5051325919589392
number of test data: 40932
MR   MRR hit@1    hit@3   hit@10
291.0074513827812	0.3271437537342339	0.23944591029023746	0.3603048959249487	0.5006351998436431
epoch: 54, lr: 8.1450625e-05 ,total loss :7522.1839871406555, time: 122.66427373886108
epoch: 55, lr: 7.737809374999998e-05 ,total loss :7507.578639268875, time: 26.9100821018219
epoch: 56, lr: 7.350918906249998e-05 ,total loss :7482.1368997097015, time: 26.900330781936646
epoch: 57, lr: 6.983372960937497e-05 ,total loss :7466.597718238831, time: 26.901387929916382
epoch: 58, lr: 6.634204312890623e-05 ,total loss :7446.78262257576, time: 27.06880235671997
number of test data: 35070
MR   MRR hit@1    hit@3   hit@10
276.4357855717137	0.3283431543446852	0.24000570287995437	0.3607356715141146	0.5076418591388652
number of test data: 40932
MR   MRR hit@1    hit@3   hit@10
292.1453630411414	0.32670122579736466	0.2385419720512069	0.35942538844913513	0.5015391380826737
epoch: 59, lr: 6.30249409724609e-05 ,total loss :7433.419765710831, time: 122.79240822792053
epoch: 60, lr: 0.0001 ,total loss :7496.446986436844, time: 26.778976917266846
epoch: 61, lr: 9.5e-05 ,total loss :7488.329827308655, time: 26.936355590820312
epoch: 62, lr: 9.025e-05 ,total loss :7467.512121677399, time: 26.998331308364868
epoch: 63, lr: 8.573749999999999e-05 ,total loss :7450.106869220734, time: 26.971580982208252
number of test data: 35070
MR   MRR hit@1    hit@3   hit@10
281.50068434559455	0.3281030024822581	0.24003421727972626	0.36070715711434276	0.5065297975477616
number of test data: 40932
MR   MRR hit@1    hit@3   hit@10
298.0422652203655	0.3269060508792339	0.2402765562396169	0.3579351118928955	0.5020521841102316
epoch: 64, lr: 8.1450625e-05 ,total loss :7434.957327604294, time: 122.74502325057983
epoch: 65, lr: 7.737809374999998e-05 ,total loss :7417.1463503837585, time: 26.77143383026123
epoch: 66, lr: 7.350918906249998e-05 ,total loss :7394.119843244553, time: 26.933417081832886
epoch: 67, lr: 6.983372960937497e-05 ,total loss :7383.242560386658, time: 26.89551591873169
epoch: 68, lr: 6.634204312890623e-05 ,total loss :7365.403049707413, time: 27.205711603164673
number of test data: 35070
MR   MRR hit@1    hit@3   hit@10
284.1082121471343	0.32932850147522436	0.24214428286284573	0.36050755631593956	0.5071571143427431
number of test data: 40932
MR   MRR hit@1    hit@3   hit@10
298.99188898661197	0.3271166039286979	0.23925046418450113	0.3605492035571191	0.5021010456366657
epoch: 69, lr: 6.30249409724609e-05 ,total loss :7356.695111751556, time: 123.05036211013794
epoch: 70, lr: 0.0001 ,total loss :7418.28530216217, time: 26.824867248535156
epoch: 71, lr: 9.5e-05 ,total loss :7411.665605306625, time: 26.892565727233887
epoch: 72, lr: 9.025e-05 ,total loss :7395.311799049377, time: 27.059374809265137
epoch: 73, lr: 8.573749999999999e-05 ,total loss :7382.018724679947, time: 26.94883131980896
number of test data: 35070
MR   MRR hit@1    hit@3   hit@10
286.10390647276876	0.32880466697157096	0.24103222127174223	0.3619332763045338	0.5045623039635015
number of test data: 40932
MR   MRR hit@1    hit@3   hit@10
300.7113016710642	0.3254842653349869	0.23719828007426952	0.3598407114238249	0.4991937848138376
epoch: 74, lr: 8.1450625e-05 ,total loss :7362.75560259819, time: 122.60743856430054
epoch: 75, lr: 7.737809374999998e-05 ,total loss :7347.391959428787, time: 26.8566677570343
epoch: 76, lr: 7.350918906249998e-05 ,total loss :7333.59530878067, time: 26.888216018676758
epoch: 77, lr: 6.983372960937497e-05 ,total loss :7314.766712188721, time: 26.90108847618103
epoch: 78, lr: 6.634204312890623e-05 ,total loss :7296.879548072815, time: 26.912691593170166
number of test data: 35070
MR   MRR hit@1    hit@3   hit@10
288.74941545480465	0.32780145054498316	0.23997718848018249	0.36110635871114916	0.5051325919589392
number of test data: 40932
MR   MRR hit@1    hit@3   hit@10
303.3601094498192	0.32586886332531373	0.23754031075930812	0.3609645265318088	0.501710153425193
epoch: 79, lr: 6.30249409724609e-05 ,total loss :7288.6229474544525, time: 122.95654964447021
epoch: 80, lr: 0.0001 ,total loss :7360.052495241165, time: 26.896677017211914
epoch: 81, lr: 9.5e-05 ,total loss :7345.116460323334, time: 26.916815757751465
epoch: 82, lr: 9.025e-05 ,total loss :7332.040389060974, time: 27.039366722106934
epoch: 83, lr: 8.573749999999999e-05 ,total loss :7320.806123256683, time: 27.022757530212402
number of test data: 35070
MR   MRR hit@1    hit@3   hit@10
290.23401767892784	0.3273020112886003	0.24009124607927004	0.3584830339321357	0.5052466495580268
number of test data: 40932
MR   MRR hit@1    hit@3   hit@10
305.0617365386495	0.32500629902809597	0.23732043389035473	0.35646926609987295	0.5021254763998827
epoch: 84, lr: 8.1450625e-05 ,total loss :7307.79342675209, time: 122.6108009815216
epoch: 85, lr: 7.737809374999998e-05 ,total loss :7287.028906106949, time: 26.882928133010864
epoch: 86, lr: 7.350918906249998e-05 ,total loss :7276.212584257126, time: 26.845125198364258
epoch: 87, lr: 6.983372960937497e-05 ,total loss :7263.177803039551, time: 26.89669680595398
epoch: 88, lr: 6.634204312890623e-05 ,total loss :7250.273107767105, time: 27.137247800827026
number of test data: 35070
MR   MRR hit@1    hit@3   hit@10
292.9370402053037	0.32871870550936416	0.24086113487311092	0.3626746506986028	0.5045337895637296
number of test data: 40932
MR   MRR hit@1    hit@3   hit@10
306.65244796247435	0.32510770998209176	0.23680738786279684	0.3599628652399101	0.5013681227401544
epoch: 89, lr: 6.30249409724609e-05 ,total loss :7239.075999736786, time: 122.85480308532715
epoch: 90, lr: 0.0001 ,total loss :7298.648550748825, time: 26.826530694961548
epoch: 91, lr: 9.5e-05 ,total loss :7294.129781246185, time: 26.887122631072998
epoch: 92, lr: 9.025e-05 ,total loss :7286.9443192481995, time: 26.86251950263977
epoch: 93, lr: 8.573749999999999e-05 ,total loss :7268.872519016266, time: 26.914955139160156
number of test data: 35070
MR   MRR hit@1    hit@3   hit@10
294.0529797547762	0.3263670359087241	0.2388366124893071	0.3587396635300827	0.5025948103792415
number of test data: 40932
MR   MRR hit@1    hit@3   hit@10
307.4806996970585	0.32461772503186104	0.23663637252027753	0.3576908042607251	0.5001465845793023
epoch: 94, lr: 8.1450625e-05 ,total loss :7259.255127668381, time: 122.6177875995636
epoch: 95, lr: 7.737809374999998e-05 ,total loss :7242.062842845917, time: 26.856398582458496
epoch: 96, lr: 7.350918906249998e-05 ,total loss :7228.9501214027405, time: 26.830010175704956
epoch: 97, lr: 6.983372960937497e-05 ,total loss :7216.219879388809, time: 26.895326852798462
epoch: 98, lr: 6.634204312890623e-05 ,total loss :7206.2676339149475, time: 26.98194432258606
number of test data: 35070
MR   MRR hit@1    hit@3   hit@10
296.1929284288566	0.32799207456089347	0.2407470772740234	0.3615625891074993	0.5028514399771885
number of test data: 40932
MR   MRR hit@1    hit@3   hit@10
308.203288380729	0.3244873182052051	0.2365142187041923	0.3590589270008795	0.5014658457930226
epoch: 99, lr: 6.30249409724609e-05 ,total loss :7192.711644411087, time: 123.08050894737244
head ranking ....
MR   MRR hit@1    hit@3   hit@10
398.33890354734683	0.2252068754824893	0.14467897977132807	0.2484120003908922	0.3905990423140819
tail ranking ....
MR   MRR hit@1    hit@3   hit@10
218.06772207563765	0.423767760935049	0.3283494576370566	0.4697058536108668	0.6123326492719633
Total ranking ...
MR   MRR hit@1    hit@3   hit@10
308.20331281149225	0.32448731820879223	0.2365142187041923	0.3590589270008795	0.5014658457930226

number of test data: 40932
MR   MRR hit@1    hit@3   hit@10
317.63607935111895	0.3194826029037716	0.23255643506303136	0.3534642822241767	0.49284178637740644
epoch: 179, lr: 6.30249409724609e-05 ,total loss :7017.966312885284, time: 124.34840416908264
epoch: 180, lr: 0.0001 ,total loss :7070.369735002518, time: 26.9829740524292
epoch: 181, lr: 9.5e-05 ,total loss :7077.173418998718, time: 26.99249815940857
epoch: 182, lr: 9.025e-05 ,total loss :7066.299565792084, time: 27.130272150039673
epoch: 183, lr: 8.573749999999999e-05 ,total loss :7054.508033752441, time: 27.087394952774048
number of test data: 35070
MR   MRR hit@1    hit@3   hit@10
303.29458226404336	0.3223760253562001	0.23510122611919018	0.35640148274878813	0.4939549472483604
number of test data: 40932
MR   MRR hit@1    hit@3   hit@10
320.0325906381315	0.31771552019866367	0.23133489690217923	0.34884686797615555	0.4911072021889964
epoch: 184, lr: 8.1450625e-05 ,total loss :7050.807545900345, time: 124.95017170906067
epoch: 185, lr: 7.737809374999998e-05 ,total loss :7039.334016561508, time: 26.965521812438965
epoch: 186, lr: 7.350918906249998e-05 ,total loss :7028.977983951569, time: 27.0393545627594
epoch: 187, lr: 6.983372960937497e-05 ,total loss :7020.3039021492, time: 27.0311598777771
epoch: 188, lr: 6.634204312890623e-05 ,total loss :7012.682553052902, time: 27.139700889587402
number of test data: 35070
MR   MRR hit@1    hit@3   hit@10
304.53390362132876	0.32391796586744126	0.23795266609637866	0.35568862275449104	0.49469632164242944
number of test data: 40932
MR   MRR hit@1    hit@3   hit@10
320.95949379458614	0.31822731732161613	0.23135932766539627	0.3499462523209225	0.49159581745333725
epoch: 189, lr: 6.30249409724609e-05 ,total loss :7001.391878128052, time: 124.50017976760864
epoch: 190, lr: 0.0001 ,total loss :7061.042743444443, time: 27.009299993515015
epoch: 191, lr: 9.5e-05 ,total loss :7056.891512155533, time: 27.01949453353882
epoch: 192, lr: 9.025e-05 ,total loss :7051.041465520859, time: 27.040632724761963
epoch: 193, lr: 8.573749999999999e-05 ,total loss :7045.551285982132, time: 27.1126766204834
number of test data: 35070
MR   MRR hit@1    hit@3   hit@10
303.24610778443116	0.32255520316085456	0.2364128885086969	0.35469061876247504	0.49389791844881664
number of test data: 40932
MR   MRR hit@1    hit@3   hit@10
319.2501221538161	0.3181268266027195	0.23118831232287695	0.3499462523209225	0.4915713866901202
epoch: 194, lr: 8.1450625e-05 ,total loss :7033.299690485001, time: 124.55558466911316
epoch: 195, lr: 7.737809374999998e-05 ,total loss :7019.728657722473, time: 27.009089946746826
epoch: 196, lr: 7.350918906249998e-05 ,total loss :7015.1301963329315, time: 27.340009212493896
epoch: 197, lr: 6.983372960937497e-05 ,total loss :7006.60594534874, time: 28.084837913513184
epoch: 198, lr: 6.634204312890623e-05 ,total loss :6998.607264757156, time: 28.219569444656372
number of test data: 35070
MR   MRR hit@1    hit@3   hit@10
304.8451382948389	0.32280752739740914	0.23664100370687197	0.35543199315654406	0.4937553464499572
number of test data: 40932
MR   MRR hit@1    hit@3   hit@10
320.459029610085	0.3180298794220433	0.2305775432424509	0.35143652887716215	0.49115606371543047
epoch: 199, lr: 6.30249409724609e-05 ,total loss :6986.808637857437, time: 131.15496468544006
head ranking ....
MR   MRR hit@1    hit@3   hit@10
415.3975373790677	0.22200095708096704	0.14184501123815108	0.24469852438190168	0.3823414443467214
tail ranking ....
MR   MRR hit@1    hit@3   hit@10
225.52042411804945	0.41405880222678165	0.3193100752467507	0.45817453337242253	0.5999706830841396
Total ranking ...
MR   MRR hit@1    hit@3   hit@10
320.4589807485586	0.31802987965389595	0.2305775432424509	0.35143652887716215	0.49115606371543047

datasets/FB15k-237/entity2vec.txt vector size:(14541, 100)
datasets/FB15k-237/relation2vec.txt vector size:(237, 100)
data num: 35070
data num: 40932
data num: 544230
init embedding from pretrained...
epoch: 0, lr: 1.8813804334965088e-08 ,total loss :20513.072697639465, time: 68.59006690979004
epoch: 1, lr: 4.0016961820470746e-05 ,total loss :15551.729062080383, time: 55.07742881774902
epoch: 2, lr: 8.001510983660652e-05 ,total loss :13105.804226398468, time: 55.13215136528015
epoch: 3, lr: 0.00012001325785274231 ,total loss :12091.7609333992, time: 55.08366417884827
number of test data: 35070
MR   MRR hit@1    hit@3   hit@10
286.56030795551754	0.2959019749899474	0.21593954947248362	0.3234103222127174	0.4550898203592814
number of test data: 40932
MR   MRR hit@1    hit@3   hit@10
298.8256620736832	0.29370097014528407	0.2119613016710642	0.3235854588097332	0.45343496530831623
epoch: 4, lr: 0.00016001140586887808 ,total loss :11426.172776222229, time: 156.34720134735107
epoch: 5, lr: 0.00015475618749999996 ,total loss :10971.965666294098, time: 55.10735726356506
epoch: 6, lr: 0.00014701837812499996 ,total loss :10662.659840106964, time: 55.06135654449463
epoch: 7, lr: 0.00013966745921874994 ,total loss :10446.82015132904, time: 55.06649112701416
epoch: 8, lr: 0.00013268408625781246 ,total loss :10285.612536907196, time: 55.04465174674988
number of test data: 35070
MR   MRR hit@1    hit@3   hit@10
235.3945252352438	0.3178911415474219	0.2345024237239806	0.3470772740233818	0.483404619332763
number of test data: 40932
MR   MRR hit@1    hit@3   hit@10
252.27213427147464	0.3162326126482889	0.23236098895729504	0.34586631486367636	0.48282517345841885
epoch: 9, lr: 0.0001260498819449218 ,total loss :10155.79657125473, time: 155.64885807037354
epoch: 10, lr: 0.00011974738784767575 ,total loss :10052.040225028992, time: 54.76395559310913
epoch: 11, lr: 0.00011376001845529194 ,total loss :9964.611527442932, time: 54.66166710853577
epoch: 12, lr: 0.00010807201753252734 ,total loss :9884.838072776794, time: 54.72834849357605
epoch: 13, lr: 0.00010266841665590097 ,total loss :9816.77418088913, time: 54.72475337982178
number of test data: 35070
MR   MRR hit@1    hit@3   hit@10
222.23321927573426	0.32613209275581	0.24100370687197034	0.35799828913601367	0.4942686056458512
number of test data: 40932
MR   MRR hit@1    hit@3   hit@10
242.03952897488517	0.32359272745002926	0.23754031075930812	0.3557119124401446	0.49369686309000294
epoch: 14, lr: 9.75349958231059e-05 ,total loss :9763.069995164871, time: 154.6115288734436
epoch: 15, lr: 9.265824603195061e-05 ,total loss :9709.092350959778, time: 55.074546337127686
epoch: 16, lr: 8.802533373035307e-05 ,total loss :9664.906946182251, time: 54.99826741218567
epoch: 17, lr: 8.362406704383542e-05 ,total loss :9620.538283586502, time: 55.04712176322937
epoch: 18, lr: 7.944286369164364e-05 ,total loss :9587.120221614838, time: 55.02715349197388
number of test data: 35070
MR   MRR hit@1    hit@3   hit@10
216.39210151126318	0.33065794749291416	0.24613629883090962	0.3600513259195894	0.4991445680068435
number of test data: 40932
MR   MRR hit@1    hit@3   hit@10
236.0663539528975	0.3278732480968756	0.24162024821655428	0.3594498192123522	0.49921821557705465
epoch: 19, lr: 7.547072050706145e-05 ,total loss :9544.734806776047, time: 155.12067818641663
epoch: 20, lr: 7.169718448170838e-05 ,total loss :9513.721664905548, time: 54.722718715667725
epoch: 21, lr: 6.811232525762296e-05 ,total loss :9488.904676437378, time: 54.70733046531677
epoch: 22, lr: 6.470670899474181e-05 ,total loss :9458.568810462952, time: 54.7235951423645
epoch: 23, lr: 6.147137354500472e-05 ,total loss :9429.751175880432, time: 54.70194888114929
number of test data: 35070
MR   MRR hit@1    hit@3   hit@10
214.91305959509552	0.3332695328631812	0.24781864841745083	0.364613629883091	0.5029654975762761
number of test data: 40932
MR   MRR hit@1    hit@3   hit@10
233.770937164077	0.33049883344616704	0.2436235708003518	0.36267467995700187	0.5014902765562396
epoch: 24, lr: 5.839780486775448e-05 ,total loss :9409.004209041595, time: 154.89324712753296
epoch: 25, lr: 5.5477914624366753e-05 ,total loss :9390.883186101913, time: 55.08030915260315
epoch: 26, lr: 5.270401889314841e-05 ,total loss :9370.821925640106, time: 55.00497317314148
epoch: 27, lr: 5.0068817948490985e-05 ,total loss :9346.841778993607, time: 55.04723310470581
epoch: 28, lr: 6.634204312890623e-05 ,total loss :9350.318042039871, time: 55.00456476211548
number of test data: 35070
MR   MRR hit@1    hit@3   hit@10
214.05702879954376	0.3330894676156827	0.24704875962360992	0.3642999714856002	0.5037353863701169
number of test data: 40932
MR   MRR hit@1    hit@3   hit@10
233.17377601876282	0.33167942415027984	0.24482067819798692	0.36421381803967556	0.5031271376917815
epoch: 29, lr: 6.30249409724609e-05 ,total loss :9335.193590641022, time: 154.93369364738464
epoch: 30, lr: 0.0001 ,total loss :9359.71115231514, time: 54.73429775238037
epoch: 31, lr: 9.5e-05 ,total loss :9346.87385392189, time: 54.66287922859192
epoch: 32, lr: 9.025e-05 ,total loss :9331.090392827988, time: 54.78004240989685
epoch: 33, lr: 8.573749999999999e-05 ,total loss :9305.492251873016, time: 54.74559235572815
number of test data: 35070
MR   MRR hit@1    hit@3   hit@10
212.0968634160251	0.3354559346244995	0.24947248360422014	0.368320501853436	0.5047904191616767
number of test data: 40932
MR   MRR hit@1    hit@3   hit@10
231.64370174924264	0.33298292633397486	0.24599335483240498	0.3651421870419232	0.5049594449330597
epoch: 34, lr: 8.1450625e-05 ,total loss :9297.517037391663, time: 154.00762391090393
epoch: 35, lr: 7.737809374999998e-05 ,total loss :9270.393520593643, time: 55.19163250923157
epoch: 36, lr: 7.350918906249998e-05 ,total loss :9248.91608786583, time: 55.05674481391907
epoch: 37, lr: 6.983372960937497e-05 ,total loss :9231.70573759079, time: 55.0942816734314
epoch: 38, lr: 6.634204312890623e-05 ,total loss :9216.981722354889, time: 55.08730864524841
number of test data: 35070
MR   MRR hit@1    hit@3   hit@10
209.74271457085828	0.33803106559909063	0.2521243227830054	0.370544625035643	0.5087824351297405
number of test data: 40932
MR   MRR hit@1    hit@3   hit@10
228.3345793022574	0.33497874944581624	0.24745920062542753	0.36768298641649566	0.5082820287305776
epoch: 39, lr: 6.30249409724609e-05 ,total loss :9200.649610042572, time: 154.25857734680176
epoch: 40, lr: 0.0001 ,total loss :9233.699474811554, time: 54.83518600463867
epoch: 41, lr: 9.5e-05 ,total loss :9219.986332654953, time: 54.726277351379395
epoch: 42, lr: 9.025e-05 ,total loss :9200.701162576675, time: 54.67863869667053
epoch: 43, lr: 8.573749999999999e-05 ,total loss :9184.749069452286, time: 54.716397285461426
number of test data: 35070
MR   MRR hit@1    hit@3   hit@10
209.52055888223552	0.3368446146888798	0.25018534359851724	0.36946107784431137	0.5088964927288281
number of test data: 40932
MR   MRR hit@1    hit@3   hit@10
228.6955438287892	0.3352445887770748	0.2480699697058536	0.3685869246555262	0.5075491058340662
epoch: 44, lr: 8.1450625e-05 ,total loss :9174.723707437515, time: 154.44892692565918
epoch: 45, lr: 7.737809374999998e-05 ,total loss :9151.922768592834, time: 55.17004466056824
epoch: 46, lr: 7.350918906249998e-05 ,total loss :9137.253393650055, time: 55.053510904312134
epoch: 47, lr: 6.983372960937497e-05 ,total loss :9127.386873483658, time: 54.99295616149902
epoch: 48, lr: 6.634204312890623e-05 ,total loss :9115.706072330475, time: 55.03166723251343
number of test data: 35070
MR   MRR hit@1    hit@3   hit@10
208.75195323638437	0.33803296597651866	0.2513544339891645	0.37142857142857144	0.510635871114913
number of test data: 40932
MR   MRR hit@1    hit@3   hit@10
227.64734193296198	0.33610778473474373	0.24870516954949673	0.3701260627381999	0.5087706439949184
epoch: 49, lr: 6.30249409724609e-05 ,total loss :9097.274609327316, time: 154.66385507583618
epoch: 50, lr: 0.0001 ,total loss :9134.825513124466, time: 54.836650133132935
epoch: 51, lr: 9.5e-05 ,total loss :9125.109899282455, time: 54.68433594703674
epoch: 52, lr: 9.025e-05 ,total loss :9111.448443889618, time: 54.69780087471008
epoch: 53, lr: 8.573749999999999e-05 ,total loss :9099.642574548721, time: 54.92047643661499
number of test data: 35070
MR   MRR hit@1    hit@3   hit@10
207.98152266894783	0.3371974600473018	0.24961505560307956	0.37057313943541487	0.5100370687197034
number of test data: 40932
MR   MRR hit@1    hit@3   hit@10
226.88673898172578	0.3357760642358436	0.2477768005472491	0.37029707808071927	0.5100899052086387
epoch: 54, lr: 8.1450625e-05 ,total loss :9087.18514585495, time: 154.4146864414215
epoch: 55, lr: 7.737809374999998e-05 ,total loss :9072.302423477173, time: 55.1620352268219
epoch: 56, lr: 7.350918906249998e-05 ,total loss :9057.58427286148, time: 55.10998010635376
epoch: 57, lr: 6.983372960937497e-05 ,total loss :9050.229278802872, time: 55.18097472190857
epoch: 58, lr: 6.634204312890623e-05 ,total loss :9031.40291929245, time: 55.126834869384766
number of test data: 35070
MR   MRR hit@1    hit@3   hit@10
208.2224123182207	0.3385319706430139	0.2504419731964642	0.3735671514114628	0.5119190191046479
number of test data: 40932
MR   MRR hit@1    hit@3   hit@10
225.79480601974007	0.3369349925288002	0.2493648001563569	0.37037037037037035	0.5084286133098798
epoch: 59, lr: 6.30249409724609e-05 ,total loss :9024.598451375961, time: 155.68406772613525
epoch: 60, lr: 0.0001 ,total loss :9061.137352228165, time: 54.9389865398407
epoch: 61, lr: 9.5e-05 ,total loss :9045.681410074234, time: 54.696341037750244
epoch: 62, lr: 9.025e-05 ,total loss :9047.299577236176, time: 54.650280714035034
epoch: 63, lr: 8.573749999999999e-05 ,total loss :9033.43666601181, time: 54.68994092941284
number of test data: 35070
MR   MRR hit@1    hit@3   hit@10
208.38716852010265	0.33876776075673104	0.25158254918733963	0.372027373823781	0.5111776447105788
number of test data: 40932
MR   MRR hit@1    hit@3   hit@10
225.42392260334213	0.3372106608756185	0.24973126160461254	0.3698573243428125	0.5102120590247239
epoch: 64, lr: 8.1450625e-05 ,total loss :9021.746648073196, time: 154.84954500198364
epoch: 65, lr: 7.737809374999998e-05 ,total loss :9007.327780723572, time: 55.21763253211975
epoch: 66, lr: 7.350918906249998e-05 ,total loss :8991.764838695526, time: 55.129703521728516
epoch: 67, lr: 6.983372960937497e-05 ,total loss :8984.727674245834, time: 55.12411665916443
epoch: 68, lr: 6.634204312890623e-05 ,total loss :8968.715107679367, time: 55.13608169555664
number of test data: 35070
MR   MRR hit@1    hit@3   hit@10
206.81177644710579	0.33929193868449325	0.2520102651839179	0.37214143142286854	0.5132591958939264
number of test data: 40932
MR   MRR hit@1    hit@3   hit@10
225.00092836900225	0.337210887598911	0.2489739079448842	0.37120101631974983	0.5110182742108863
epoch: 69, lr: 6.30249409724609e-05 ,total loss :8966.818740606308, time: 155.44690012931824
epoch: 70, lr: 0.0001 ,total loss :9000.341579675674, time: 54.82669115066528
epoch: 71, lr: 9.5e-05 ,total loss :8994.0391330719, time: 54.705583810806274
epoch: 72, lr: 9.025e-05 ,total loss :8987.944967031479, time: 54.714715242385864
epoch: 73, lr: 8.573749999999999e-05 ,total loss :8980.922403335571, time: 54.76743221282959
number of test data: 35070
MR   MRR hit@1    hit@3   hit@10
208.7589392643285	0.33869347443187686	0.2510122611919019	0.37359566581123466	0.5135158254918734
number of test data: 40932
MR   MRR hit@1    hit@3   hit@10
226.22916055897585	0.33673490179597004	0.2484608619173263	0.37171406234730775	0.509723443760383
epoch: 74, lr: 8.1450625e-05 ,total loss :8965.079819202423, time: 154.37773084640503
epoch: 75, lr: 7.737809374999998e-05 ,total loss :8958.506523609161, time: 55.07911658287048
epoch: 76, lr: 7.350918906249998e-05 ,total loss :8946.78271484375, time: 55.03142809867859
epoch: 77, lr: 6.983372960937497e-05 ,total loss :8936.918813467026, time: 55.16730499267578
epoch: 78, lr: 6.634204312890623e-05 ,total loss :8922.185603380203, time: 55.10928559303284
number of test data: 35070
MR   MRR hit@1    hit@3   hit@10
208.16278870829768	0.3388281798691715	0.25109780439121754	0.3726261762189906	0.5138009694895922
number of test data: 40932
MR   MRR hit@1    hit@3   hit@10
225.00373790677222	0.3372686367621488	0.24899833870810123	0.3705658164761067	0.5103586436040262
epoch: 79, lr: 6.30249409724609e-05 ,total loss :8911.760321855545, time: 154.64064264297485
epoch: 80, lr: 0.0001 ,total loss :8947.926607608795, time: 54.81027674674988
epoch: 81, lr: 9.5e-05 ,total loss :8945.083136320114, time: 54.81056022644043
epoch: 82, lr: 9.025e-05 ,total loss :8942.731804132462, time: 54.79671382904053
epoch: 83, lr: 8.573749999999999e-05 ,total loss :8934.630988359451, time: 54.76520490646362
number of test data: 35070
MR   MRR hit@1    hit@3   hit@10
208.12192757342459	0.3389261958710537	0.25098374679213004	0.37359566581123466	0.5137154262902766
number of test data: 40932
MR   MRR hit@1    hit@3   hit@10
225.7293804358448	0.336114023555634	0.24782566207368317	0.3698817551060295	0.5103342128408092
epoch: 84, lr: 8.1450625e-05 ,total loss :8923.84552359581, time: 154.3715283870697
epoch: 85, lr: 7.737809374999998e-05 ,total loss :8914.63568019867, time: 55.20500040054321
epoch: 86, lr: 7.350918906249998e-05 ,total loss :8902.547950267792, time: 55.060322761535645
epoch: 87, lr: 6.983372960937497e-05 ,total loss :8896.863480567932, time: 55.08317995071411
epoch: 88, lr: 6.634204312890623e-05 ,total loss :8881.076248884201, time: 55.04885292053223
number of test data: 35070
MR   MRR hit@1    hit@3   hit@10
207.19435414884518	0.3397930001509046	0.25297975477616197	0.3727972626176219	0.5132306814941545
number of test data: 40932
MR   MRR hit@1    hit@3   hit@10
225.05123131046614	0.337487143207677	0.24946252320922505	0.37159190853122254	0.5097723052868172
epoch: 89, lr: 6.30249409724609e-05 ,total loss :8877.228162050247, time: 154.58804893493652
epoch: 90, lr: 0.0001 ,total loss :8906.102368354797, time: 54.69085121154785
epoch: 91, lr: 9.5e-05 ,total loss :8908.299256324768, time: 54.59702157974243
epoch: 92, lr: 9.025e-05 ,total loss :8902.694072246552, time: 54.66360640525818
epoch: 93, lr: 8.573749999999999e-05 ,total loss :8896.702796936035, time: 54.65752100944519
number of test data: 35070
MR   MRR hit@1    hit@3   hit@10
207.4581123467351	0.33924941416819143	0.2517821499857428	0.3722269746221842	0.5135728542914172
number of test data: 40932
MR   MRR hit@1    hit@3   hit@10
225.51050522818332	0.337285113300482	0.24973126160461254	0.36993061663246357	0.510285351314375
epoch: 94, lr: 8.1450625e-05 ,total loss :8886.287888288498, time: 154.85961365699768
epoch: 95, lr: 7.737809374999998e-05 ,total loss :8877.733994483948, time: 55.01132655143738
epoch: 96, lr: 7.350918906249998e-05 ,total loss :8867.955391645432, time: 55.02659368515015
epoch: 97, lr: 6.983372960937497e-05 ,total loss :8860.58357667923, time: 55.01776432991028
epoch: 98, lr: 6.634204312890623e-05 ,total loss :8850.764257192612, time: 55.08710336685181
number of test data: 35070
MR   MRR hit@1    hit@3   hit@10
207.72358140861135	0.3407039303724182	0.2538066723695466	0.37276874821785	0.5146278870829769
number of test data: 40932
MR   MRR hit@1    hit@3   hit@10
225.276678393433	0.33815153653637015	0.25031759992182157	0.3721049545587804	0.51082282810515
epoch: 99, lr: 6.30249409724609e-05 ,total loss :8842.36862897873, time: 154.66338634490967
head ranking ....
MR   MRR hit@1    hit@3   hit@10
302.3322583797518	0.23696750743165812	0.1541092543731066	0.26214208931887034	0.4015440242353171
tail ranking ....
MR   MRR hit@1    hit@3   hit@10
148.22104954558782	0.4393355656733523	0.3465259454705365	0.4820678197986905	0.6201016319749829
Total ranking ...
MR   MRR hit@1    hit@3   hit@10
225.2766539626698	0.33815153655253155	0.25031759992182157	0.3721049545587804	0.51082282810515